swap/swap.go

```diff
@@ -158,8 +158,9 @@ func (s *Swap) Add(amount int64, peer *protocols.Peer) (err error) {
 	if !exists {
 		return fmt.Errorf("peer %v does not exist", peer.ID())
 	}
-	if balance >= s.disconnectThreshold {
-		return fmt.Errorf("balance for peer %s is over the disconnect threshold %d, disconnecting", peer.ID().String(), s.disconnectThreshold)
+	disconnectThreshold := s.disconnectThreshold
+	if balance >= int64(disconnectThreshold) {
+		return fmt.Errorf("balance for peer %s is over the disconnect threshold %d, disconnecting", peer.ID().String(), disconnectThreshold)
```

Comment:
do this instead:
```
disconnectThreshold := int64(s.disconnectThreshold)
if balance >= disconnectThreshold { ...
```
this way, the `Errorf` call receives the same parameter as the comparison that triggers it.